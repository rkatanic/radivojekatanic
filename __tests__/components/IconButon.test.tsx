import IconButon from "@/components/IconButton";
import { render } from "@testing-library/react";

describe("IconButon", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <IconButon href="link" icon={<div>icon</div>} />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
