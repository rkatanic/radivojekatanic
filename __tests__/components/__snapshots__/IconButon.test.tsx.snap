// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButon should render 1`] = `
<body>
  <div>
    <button
      class="group relative flex h-10 w-10 items-center justify-center border-2 border-neutral-200 bg-slate-50 shadow-sm hover:border-sky-500 hover:bg-sky-500 dark:border-neutral-800 dark:bg-neutral-900 dark:shadow-md dark:hover:border-sky-500 dark:hover:bg-sky-500"
      type="button"
    >
      <div>
        icon
      </div>
      <div
        class="pointer-events-none absolute left-[3.25rem] whitespace-nowrap border-2 border-neutral-300 bg-slate-50 px-2 py-0.5 text-sm font-medium text-neutral-500 opacity-0 shadow-sm transition-opacity duration-300 group-hover:opacity-100 dark:border-neutral-700 dark:bg-neutral-800 dark:text-neutral-300 dark:shadow-xl"
      >
        <svg
          class="absolute -left-2.5 top-1/2 -translate-y-1/2 fill-neutral-300 dark:fill-neutral-600"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 6 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 2L0 8l6 6V2z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </button>
  </div>
</body>
`;
